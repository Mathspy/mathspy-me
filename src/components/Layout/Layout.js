import React from "react";
import { css } from "emotion";
import Helmet from "react-helmet";

import NavBar from "../NavBar";

import MathspyIcon from "../static/MathspyIcon";
import ProjectsIcon from "../static/ProjectsIcon";
import BlogIcon from "../static/BlogIcon";

const Layout = ({ data, children }) => (
  <>
    <Helmet
      title={data.site.siteMetadata.title}
      meta={[
        { name: "description", content: "Mathspy's personal website" },
        { name: "keywords", content: "mathspy, blog" },
      ]}
    />
    <div className={layoutStyles}>
      <NavBar>
        <MathspyIcon />
        <ProjectsIcon />
        <BlogIcon />
      </NavBar>
      <div className={contentStyles}>{children}</div>
    </div>
  </>
);

const layoutStyles = css`
  display: flex;
  flex-direction: column;
  min-height: 100vh;

  @media (min-width: 420px) {
    flex-direction: row;
  }
`;

const contentStyles = css`
  @media (min-width: 900px) {
    flex: 1 768px;
  }
`;

export default Layout;
