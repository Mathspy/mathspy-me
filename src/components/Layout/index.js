import React from "react";
import PropTypes from "prop-types";
import { StaticQuery, graphql } from "gatsby";

//Global styles
import "../../css/index.js";

import InnerLayout from "./Layout";

const Layout = props => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
          }
        }
      }
    `}
    render={data => <InnerLayout data={data} {...props} />}
  />
);

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
