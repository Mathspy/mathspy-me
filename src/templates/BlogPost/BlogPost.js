import React from "react";

import { css } from "emotion";

import theme from "../../theme";
import prismTheme from "../../prism-theme";

const BlogPost = ({ title, html }) => {
  return (
    <div className={`${blogStyles} ${prismTheme}`}>
      <h1>{title}</h1>
      <div dangerouslySetInnerHTML={{ __html: html }} />
    </div>
  );
};

const blogStyles = css`
  padding: 10px;

  p,
  ul > li {
    line-height: 1.7;
  }

  p > code {
    background-color: ${theme.mainColorLightened};
  }

  a {
    background-color: #f895544d;
    color: black;
    text-decoration: none;
    padding: 2px 3px;
    border-bottom: solid #0000004d 1px;
  }

  ul li {
    margin-bottom: 7px;
  }

  a:hover {
    background-color: #f89554bf;
    border-bottom-color: black;
  }
`;

export default BlogPost;
