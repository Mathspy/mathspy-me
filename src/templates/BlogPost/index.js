import React from "react";
import { graphql } from "gatsby";
import Layout from "../../components/Layout";

import BlogPost from "./BlogPost";

export default ({ data }) => {
  const { frontmatter, html } = data.markdownRemark;
  return (
    <Layout>
      <BlogPost title={frontmatter.title} html={html} />
    </Layout>
  );
};

export const query = graphql`
  query($slug: String!) {
    markdownRemark(fields: { slug: { eq: $slug } }) {
      html
      frontmatter {
        title
      }
    }
  }
`;
