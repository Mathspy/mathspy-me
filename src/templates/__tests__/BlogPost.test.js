import React from "react";

import { shallow, render } from "enzyme";

import BlogPost from "../BlogPost/BlogPost.js";
import BlogPostWrapper from "../BlogPost";

import Layout from "../../components/Layout";

const data = {
  markdownRemark: {
    html: "<h1>Test HTML</h1>",
    frontmatter: {
      title: "Markdown Test!",
    },
  },
};

describe("BlogPost", () => {
  it("should be a simple div containing an h1 and another div", () => {
    const wrapper = shallow(<BlogPost />);

    expect(wrapper.is("div")).toBe(true);
    expect(
      wrapper
        .children()
        .first()
        .is("h1")
    ).toBe(true);
    expect(
      wrapper
        .children()
        .last()
        .is("div")
    ).toBe(true);
  });

  it("should display the title passed to it", () => {
    const wrapper = shallow(<BlogPost title="testtitle" />);

    expect(wrapper.find("h1").text()).toBe("testtitle");
  });

  it("should render the html passed to it", () => {
    const wrapper = render(<BlogPost html="<div class=&quot;test&quot; />" />);

    expect(wrapper.find("div.test")).toHaveLength(1);
  });

  it("should match snapshot", () => {
    const wrapper = shallow(<BlogPost />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe("BlogPostWrapper", () => {
  it("should render a BlogPost and surround it with Layout", () => {
    const wrapper = shallow(<BlogPostWrapper data={data} />);
    expect(wrapper.is(Layout)).toBe(true);
    expect(wrapper.find(BlogPost)).toHaveLength(1);
  });

  it("should pass to it the title and html", () => {
    const wrapper = shallow(<BlogPostWrapper data={data} />);
    expect(wrapper.find(BlogPost).props()).toEqual({
      title: data.markdownRemark.frontmatter.title,
      html: data.markdownRemark.html,
    });
  });
});
