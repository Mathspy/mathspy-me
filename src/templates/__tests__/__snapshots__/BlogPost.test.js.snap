// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost should match snapshot 1`] = `
.emotion-1 code[class*="prism-"],
.emotion-1 pre[class*="prism-"] {
  font-family: Consolas,Menlo,Monaco,"Andale Mono WT","Andale Mono", "Lucida Console","Lucida Sans Typewriter","DejaVu Sans Mono", "Bitstream Vera Sans Mono","Liberation Mono","Nimbus Mono L", "Courier New",Courier,monospace;
  font-size: 14px;
  line-height: 1.375;
  direction: ltr;
  text-align: left;
  white-space: pre;
  word-spacing: normal;
  word-break: normal;
  tab-size: 4;
  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  hyphens: none;
  background: #2a2734;
  color: #afa0fe;
  white-space: pre-wrap;
}

.emotion-1 pre[class*="prism-"]::-moz-selection,
.emotion-1 pre[class*="prism-"] ::-moz-selection,
.emotion-1 code[class*="prism-"]::-moz-selection,
.emotion-1 code[class*="prism-"] ::-moz-selection {
  text-shadow: none;
  background: #6a51e6;
  color: white;
}

.emotion-1 pre[class*="prism-"]::selection,
.emotion-1 pre[class*="prism-"] ::selection,
.emotion-1 code[class*="prism-"]::selection,
.emotion-1 code[class*="prism-"] ::selection {
  text-shadow: none;
  background: #6a51e6;
  color: white;
}

.emotion-1 pre[class*="prism-"] {
  padding: 1em;
  margin: 0.5em 0;
}

.emotion-1:not(pre) > code[class*="prism-"] {
  padding: 0.1em;
  border-radius: 0.3em;
}

.emotion-1 .token.comment,
.emotion-1 .token.prolog,
.emotion-1 .token.doctype,
.emotion-1 .token.cdata {
  color: #6c6783;
}

.emotion-1 .token.punctuation {
  color: #6c6783;
}

.emotion-1 .token.namespace {
  opacity: 0.7;
}

.emotion-1 .token.tag,
.emotion-1 .token.operator,
.emotion-1 .token.number {
  color: #e09142;
}

.emotion-1 .token.property,
.emotion-1 .token.function {
  color: #c4b9fe;
}

.emotion-1 .token.tag-id,
.emotion-1 .token.selector,
.emotion-1 .token.atrule-id {
  color: #eeebff;
}

.emotion-1 code.prism-javascript,
.emotion-1 .token.attr-name {
  color: #c4b9fe;
}

.emotion-1 code.prism-css,
.emotion-1 code.prism-scss,
.emotion-1 .token.boolean,
.emotion-1 .token.string,
.emotion-1 .token.entity,
.emotion-1 .token.url,
.emotion-1 .prism-css .token.string,
.emotion-1 .prism-scss .token.string,
.emotion-1 .style .token.string,
.emotion-1 .token.attr-value,
.emotion-1 .token.keyword,
.emotion-1 .token.control,
.emotion-1 .token.directive,
.emotion-1 .token.unit,
.emotion-1 .token.statement,
.emotion-1 .token.regex,
.emotion-1 .token.atrule {
  color: #fc9;
}

.emotion-1 .token.placeholder,
.emotion-1 .token.variable {
  color: #fc9;
}

.emotion-1 .token.deleted {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.emotion-1 .token.inserted {
  border-bottom: 1px dotted #eeebff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1 .token.italic {
  font-style: italic;
}

.emotion-1 .token.important,
.emotion-1 .token.bold {
  font-weight: bold;
}

.emotion-1 .token.important {
  color: #c4b9fe;
}

.emotion-1 .token.entity {
  cursor: help;
}

.emotion-1 pre > code.highlight {
  outline: 0.4em solid #8a75f5;
  outline-offset: 0.4em;
}

.emotion-1 .line-numbers .line-numbers-rows {
  border-right-color: #2c2937;
}

.emotion-1 .line-numbers-rows > span:before {
  color: #3c3949;
}

.emotion-1 .line-highlight {
  background: rgba(224,145,66,0.2);
  background: -webkit-gradient( linear, left top, right top, color-stop(70%,rgba(224,145,66,0.2)), to(rgba(224,145,66,0)) );
  background: linear-gradient( to right, rgba(224,145,66,0.2) 70%, rgba(224,145,66,0) );
}

.emotion-0 {
  padding: 10px;
}

.emotion-0 p,
.emotion-0 ul > li {
  line-height: 1.7;
}

.emotion-0 p > code {
  background-color: #c697e24d;
}

.emotion-0 a {
  background-color: #f895544d;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 2px 3px;
  border-bottom: solid #0000004d 1px;
}

.emotion-0 ul li {
  margin-bottom: 7px;
}

.emotion-0 a:hover {
  background-color: #f89554bf;
  border-bottom-color: black;
}

<div
  className="emotion-0 emotion-1"
>
  <h1 />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
</div>
`;
